src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs

```diff
@@ -560,6 +560,18 @@ private static bool IsAsRestrictive(NamedTypeSymbol s1, Symbol sym2, ref HashSet
                             break;
                         }
 
+                    case Accessibility.ProtectedAndInternal:
+                        // Since s1 is private protected, s2 must pass the test for being both more restrictive than internal and more restrictive than protected.
+                        // We first do the "internal" test (copied from above), then if it passes we continue with the "protected" test.
+
+                        if ((acc2 == Accessibility.Private || acc2 == Accessibility.Internal || acc2 == Accessibility.ProtectedAndInternal) && s2.ContainingAssembly.HasInternalAccessTo(s1.ContainingAssembly))
```

Comment:
Would it be worth it to factor two helper methods out instead? One for `IsAsRestrictiveAsInternal` and one for `IsAsRestrictiveAsProtected`.
Then the code will be more self-explanatory.